### PR TITLE
Reduce CI false positives

### PR DESCRIPTION
**Pull request by chromatic**
I think something like this will address #3378, both the false positives in installation failures and the specific error encountered in CI.

If this approach works, we can add more guards like this around installation paths.

### CODE REVIEW

**Comment by patricklodder** [comment]
Thanks! Solves the first issue with #3378, does need a rebase on top of the current 1.21-dev (44ff0bc)

**Comment by chromatic** [comment]
Rebased now.

File: ci/test/05_before_script.sh [hunk 1]
```diff
@@ -16,6 +16,12 @@ fi
 DOCKER_EXEC mkdir -p ${DEPENDS_DIR}/SDKs ${DEPENDS_DIR}/sdk-sources
 
 DOCKER_EXEC test/functional/install-deps.sh
+retVal=$?
+if [ $retVal -eq 1 ]; then
+    echo "Error installing dependencies"
+    exit $retVal
+fi
+exit $retVal
```
Comment (patricklodder) [inline ci/test/05_before_script.sh:24]: I suspect that this `exit` currently prevents further execution of the CI and should be removed.

Comment (chromatic) [reply ci/test/05_before_script.sh:24]: Now that you mention it, I strongly suspect that this extra `exit` is suspect.

**Review by patricklodder — APPROVED** [review]
utACK - fixes false positives on the CI for installing test dependencies

**Pull request merged**